server/src/test/java/io/crate/expression/reference/doc/lucene/SourceParserTest.java

```diff
@@ -156,4 +156,61 @@ public void test_uses_inner_type_info_to_parse_objects() throws Exception {
             .extracting("bs")
             .isEqualTo(BitString.ofRawBits("1001"));
     }
+
+    @Test
+    public void test_null_object_sibling_subcolumn_has_same_name() {
+        SourceParser sourceParser = new SourceParser();
+        ObjectType innerObjectType = ObjectType.builder()
+            .setInnerType("target", DataTypes.FLOAT)
+            .build();
+        ObjectType outerObjectType = ObjectType.builder()
+            .setInnerType("inner_object", innerObjectType)
+            .setInnerType("target", DataTypes.STRING)
+            .build();
+        sourceParser.register(new ColumnIdent("_doc", List.of("outer_object")), outerObjectType);
+        Map<String, Object> result = sourceParser.parse(new BytesArray(
+            """
+            {
+                "outer_object": {
+                    "inner_object": null,
+                    "target": "some text"
+                }
+            }
+            """));
+
+        assertThat(result)
+            .extracting("outer_object")
+            .extracting("inner_object")
+            .isNull();
+
+        assertThat(result)
+            .extracting("outer_object")
+            .extracting("target")
+            .isEqualTo("some text");
+    }
+
+    @Test
+    public void test_null_object_next_not_sibling_column_has_same_name() {
```

Comment:
This case actually passes on the current master. 
Since it's a lightweight test and my previous approach used to fail on such scenario I kept it.